src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/sdl_activate_app_request.cc

```diff
@@ -186,6 +207,18 @@ void SDLActivateAppRequest::Run() {
 
   if (app_to_activate->IsRegistered()) {
     SDL_LOG_DEBUG("Application is registered. Activating.");
+    auto main_state = app_to_activate->CurrentHmiState(
+        mobile_apis::PredefinedWindows::DEFAULT_WINDOW);
+    if (mobile_apis::HMILevel::INVALID_ENUM == main_state->hmi_level()) {
+      SDL_LOG_DEBUG(
+          "Application registration is not completed, HMI level hasn't set "
+          "yet, postpone activation");
+      auto& postponed_activation_ctrl = application_manager_.state_controller()
+                                            .GetPostponedActivationController();
+      postponed_activation_ctrl.AddAppToActivate(application_id,
+                                                 correlation_id());
+      return;
+    }
     policy_handler_.OnActivateApp(application_id, correlation_id());
     return;
   } else if (app_to_activate->is_cloud_app()) {
```

Comment:
@AKalinich-Luxoft there are some cases where this delay logic will not be applied, such as if the app is a cloud app. I think the other call to `policy_handler_.OnActivateApp` in `SDLActivateAppRequest::on_event` should have the postponed_activation_ctrl logic as well.

Reply:
@iCollin fixed in https://github.com/smartdevicelink/sdl_core/pull/3557/commits/1b59bec2a67c251ed6455496501a91a64c14b4b5